Lunar/Data/Display.swift

```diff
@@ -98,19 +98,32 @@ class Display: NSManagedObject {
             observe(\.maxBrightness, options: [.new, .old], changeHandler: readapt),
             observe(\.minContrast, options: [.new, .old], changeHandler: readapt),
             observe(\.maxContrast, options: [.new, .old], changeHandler: readapt),
-            observe(\.brightness, options: [.new], changeHandler: { _, change in
+            observe(\.brightness, options: [.new, .old], changeHandler: { _, change in
                 if let newBrightness = change.newValue, self.id != GENERIC_DISPLAY_ID {
                     var brightness: UInt8
                     if brightnessAdapter.mode == AdaptiveMode.manual {
                         brightness = cap(newBrightness.uint8Value, minVal: 0, maxVal: 100)
                     } else {
                         brightness = cap(newBrightness.uint8Value, minVal: self.minBrightness.uint8Value, maxVal: self.maxBrightness.uint8Value)
                     }
-                    _ = DDC.setBrightness(for: self.id, brightness: brightness)
+					var currentValue = change.oldValue!.uint8Value
+					let incrementing = currentValue < brightness
+					let stepValue = abs(currentValue.distance(to: brightness)) >= 50 ? 2 : 1
+					func step () {
+						currentValue = self.incrOrDecrValue(incrementing, currentValue, stepValue, targetValue: brightness)
+						_ = DDC.setBrightness(for: self.id, brightness: currentValue)
+						if currentValue == brightness {
+							return
+						}
+						DispatchQueue.main.asyncAfter(deadline: DispatchTime.now()) {
```

Comment:
Can we do this without declaring `step` and `incrOrDecrValue`?
I'm thinking it would be easier to do it with a for loop inside the asyncAfter call